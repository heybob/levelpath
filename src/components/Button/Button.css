button {
    background-color: #6a6aff;
    min-width: 1rem;
    padding: 0.5rem 0.8rem;
    border-radius: 0.2rem ;
    border: 1px solid rgb(80, 80, 251);
    color: white;
    margin: .2rem;
    font-size: 1rem;
    font-weight: bold;
  }
  
  button:disabled {
    opacity: 0.5;
  }
  button:hover {
    cursor: pointer;
    transition: all 0.5s;
    background-color: #7f7ffe;
  }