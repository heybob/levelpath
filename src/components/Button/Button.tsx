import './Button.css'
import { ButtonProps } from '../../shared/types'

export function Button({key, name, clickHandler, disabled = false}: ButtonProps) {
    return (
        <button key={key} onClick={clickHandler} disabled={disabled}>{name}</button>
    )
}