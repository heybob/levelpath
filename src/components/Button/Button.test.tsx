import renderer from 'react-test-renderer';
import { Button } from "./Button";

describe('Testing Loading Indicator Component', () => {
  it('It should render a button with name ClickMe', () => {
      const tree = renderer.create(<Button name="Click Me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('It should render a disabled Click Me Button', () => {
    const tree = renderer.create(<Button name="Click Me" disabled={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
});
