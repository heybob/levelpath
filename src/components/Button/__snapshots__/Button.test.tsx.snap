// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button Component It should render a button with name ClickMe 1`] = `
<button
  disabled={false}
>
  Click Me
</button>
`;

exports[`Testing Button Component It should render a disabled Click Me Button 1`] = `
<button
  disabled={true}
>
  Click Me
</button>
`;
